crates/solver/src/driver.rs

```diff
@@ -164,14 +166,39 @@ impl Driver {
         .await
     }
 
+    /// Collects all orders which got traded in the settlement. Tapping into partially fillable
+    /// orders multiple times will not result in duplicates. Partially fillable orders get
+    /// considered as traded only the first time we tap into their liquidity.
+    fn get_traded_orders(settlement: &Settlement) -> Vec<Order> {
+        let mut traded_orders = Vec::new();
+        for (_, group) in &settlement
+            .executed_trades()
+            .group_by(|(trade, _)| trade.order.metadata.uid)
+        {
+            let mut group = group.into_iter().peekable();
+            let order = &group.peek().unwrap().0.order;
+            let previously_filled = match order.creation.kind {
+                OrderKind::Buy => order.metadata.executed_buy_amount.clone(),
+                OrderKind::Sell => order.metadata.executed_sell_amount.clone(),
+            };
+            let newly_filled = group.fold(BigUint::default(), |acc, (trade, _)| {
+                acc + u256_to_big_uint(&trade.executed_amount)
+            });
+            if previously_filled == 0u8.into() && newly_filled >= 0u8.into() {
```

Comment:
Technically, a solver could fill `0`, meaning we might count orders more than once. We should likely:
```suggestion
            if previously_filled == 0u8.into() && newly_filled > 0u8.into() {
```

Reply:
🤦‍♂️ 
Good catch!